### PR TITLE
Pages knowledge base article

### PR DESCRIPTION
**Pull request by Ephraim-G**
## Changes proposed in this pull request:
- Add a knowledge base article outlining some best practices Pages users can apply to their repositories 

## Security Considerations
N/A

### CODE REVIEW

File: _kbarticlespages/repository-best-practices.md
Comment (svenaas) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
A repository does *not need* to be public in order for Pages to be able to build the website; both public and private repositories are acceptable. When it comes to Pages there are certainly some unique considerations for each.
```

File: _kbarticlespages/repository-best-practices.md
Comment (svenaas) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
Adhering to a set of best practices when it comes to managing GitHub repositories can greatly ease the maintenance burden and drastically improve the developer experience for websites hosted on cloud.gov Pages. This document should serve as a constructive conversation piece for agencies and not as an audit mechanism. But by utilizing certain code scanning tools, agencies may find it easier to identify and cover certain security controls within their respective Customer Responsibility Matrix (CRM). Here is a useful guide for you to get started with implementing some best practices regarding user permissions and security for your GitHub hosted repositories.
```

File: _kbarticlespages/repository-best-practices.md
Comment (svenaas) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
  - Potentially out of compliance with agency/program transparency initiatives
```

File: _kbarticlespages/repository-best-practices.md
Comment (svenaas) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
It is important to designate admin/owner access of the repository to one or more individuals who are either the persistent site owners or else federal employees with the “manager” role within the site's Pages organization. It is strongly recommended to not grant admin/owner privileges of the repository to any contractors or other temporary employees. This is because in certain instances actions may need to be taken by either Pages engineers or by customers that require an active user with admin access to a website's repository. Additionally, when adding contributors or collaborators to a private repository it should mirror the member count that your organization has within Pages. Please note that removing an organization member from an organization within Pages will **not** automatically remove them from the repository within GitHub. They will still retain whatever read/write access they have in GitHub until explicitly removed.
```

File: _kbarticlespages/repository-best-practices.md
Comment (svenaas) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
Dependabot is a useful and highly configurable automated dependency management tool which assists developers with keeping third-party dependencies up-to-date with their latest version. This can benefit agencies with smaller developer teams by reducing maintenance overhead. Dependabot scans the repository for outdated dependencies and generates pull requests against whatever branch it is enabled on to update them. Check out [our knowledge base article on Dependabot](https://cloud.gov/pages/knowledge-base/dependabot-with-pages/#configuring-dependabot) to learn how to enable and configure it for your repository.
```

File: _kbarticlespages/repository-best-practices.md
Comment (svenaas) [inline _kbarticlespages/repository-best-practices.md:53]: I think "that limits users with write and push access the ability to push directly" needs to be reworded for clarity.

File: _kbarticlespages/repository-best-practices.md
Comment (svenaas) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
[Allstar](https://github.com/ossf/allstar?tab=readme-ov-file#disabling-unwanted-issues-1) is an open source security tool which monitors organizations and repositories within GitHub for adherence to security best practices and improves the security posture for those organizations. It achieves this by continuously scanning repositories for any breach of enabled [security policies](https://github.com/ossf/allstar?tab=readme-ov-file#policies) and then raising an alert for any security remediation findings. This alert can be in the form of a GitHub issue or another type of configured action. In some instances Allstar can automatically resolve the issue on the admin/owner's behalf. 
```

**Review by svenaas — COMMENTED** [review]
This is a valuable resource. I've made some copyediting suggestions, but maybe others should weigh in before decisions are made about them, in case someone offers a better wording.

**Comment by Ephraim-G** [comment]
@sknep 😀

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
## Public vs Private repositories
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
### Public
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
#### Pros
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
#### Cons
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
### Private
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
#### Pros
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
#### Cons
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
## Repository User Permissions
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
## GitHub Repository Security
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
### Additional Advanced Feature Settings for GitHub Repository Security
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
#### OpenSSF Allstar
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
#### CodeQL
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
#### Signed Commits
```

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]: Since all published sites are public, the information would  probably affect both public and private repositories. I don't want to inadvertently give a customer a false sense of security with a private repo.

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]: Public repos have the same protections. I would remove this.

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]: Like the con for a public repo, Since all published sites are public, I don't want to inadvertently give a customer a false sense of security with a private repo.

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]: This is general and not specific

**Review by apburnes — COMMENTED** [review]
Look great! I was iffy on the pros for a private repositories because all branches are published publicly and I don't want to give customers a false sense of security.

File: _kbarticlespages/repository-best-practices.md
Comment (Ephraim-G) [reply _kbarticlespages/repository-best-practices.md]: Going to rework the language so enhanced security isn't implied in any way.

File: _kbarticlespages/repository-best-practices.md
Comment (Ephraim-G) [reply _kbarticlespages/repository-best-practices.md]: Understood but if someone committed API keys to a `_config.yml` file for example in a private repository is that exposed anywhere other than in the build logs? That's more so what I was looking to convey

File: _kbarticlespages/repository-best-practices.md
Comment (Ephraim-G) [reply _kbarticlespages/repository-best-practices.md]: switched it up to explicitly state for any config files

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [reply _kbarticlespages/repository-best-practices.md]: It starts to become very specific to the static site build engine and how they use their configuration settings. Many attributes in a Jekyll `_config.yml` get included with the site output. Let's leave this `Any sensitive information inadvertently hardcoded is immediately visible` here since it is true and let's adjust the other one to be very explicit.

File: _kbarticlespages/repository-best-practices.md
Comment (apburnes) [inline _kbarticlespages/repository-best-practices.md]:
```suggestion
  - Safeguard sensitive data such as API keys, access tokens or other credentials by [using environment variables](https://cloud.gov/pages/documentation/env-vars-on-pages-builds/) in our build runtime.
```

**Review by apburnes — APPROVED** [review]

**Comment by apburnes** [comment]
Closes https://github.com/cloud-gov/product/issues/2952

**Pull request merged**
